deployment/kubernetes/aws/bookkeeper.yaml

```diff
@@ -55,13 +53,14 @@ metadata:
   name: bookie-config
 data:
   PULSAR_MEM: "\"-Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.linkCapacity=1024 -XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB -XX:+ExitOnOutOfMemoryError -XX:+PerfDisableSharedMem -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintHeapAtGC -verbosegc -XX:G1LogLevel=finest -Xms28g -Xmx28g -XX:MaxDirectMemorySize=28g\""
-  PULSAR_PREFIX_dbStorage_writeCacheMaxSizeMb: "2048" # Write cache size (direct memory)
-  PULSAR_PREFIX_dbStorage_readAheadCacheMaxSizeMb: "2048" # Read cache size (direct memory)
-  PULSAR_PREFIX_dbStorage_rocksDB_blockCacheSize: "4294967296"
-  PULSAR_PREFIX_journalMaxSizeMB: "2048"
-  PULSAR_PREFIX_zkServers: zk-0.zookeeper,zk-1.zookeeper,zk-2.zookeeper
-  PULSAR_PREFIX_statsProviderClass: org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider
-  PULSAR_PREFIX_useHostNameAsBookieID: "true"
+  BOOKIE_MEM: "\"-Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.linkCapacity=1024 -XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB -XX:+ExitOnOutOfMemoryError -XX:+PerfDisableSharedMem -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintHeapAtGC -verbosegc -XX:G1LogLevel=finest -Xms28g -Xmx28g -XX:MaxDirectMemorySize=28g\""
```

Comment:
You should also take out the garbage collector settings and put them into `BOOKIE_GC`. You would also need to add `bin/apply-config-from-env.py conf/bkenv.sh &&` in the command section if you want these changes to take effect.

Reply:
@roman-popenov What should the BOKKIE_GC be set to? I just copied BOOKIE_MEM from PULSAR_MEM in this file.
And, are you sure that bkenv.sh is needed, as just setting BOOKIE_MEM worked fine for me (bookie didn't start before I set it)? If it is needed should it be the first one called?

In addition, do you think that for the bookie-autorecovery in the same yaml the PULSAR_MEM and PULSAR_GC should be changed to BOOKIE_* as well?

Reply:
You simply would be taking out -XX:+UseG1GC -XX:MaxGCPauseMillis=10 of the BOOKIE_MEM and putting it into BOOKIE_GC like so:

`BOOKIE_MEM: "\"-Xms15g -Xmx15g -XX:MaxDirectMemorySize=15g -Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.linkCapacity=1024 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB -XX:+ExitOnOutOfMemoryError -XX:+PerfDisableSharedMem -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintHeapAtGC -verbosegc -XX:G1LogLevel=finest\""`
`BOOKIE_GC: "\"-XX:+UseG1GC -XX:MaxGCPauseMillis=10\""`

`bookie-autorecovery` does run `bin/apply-config-from-env.py conf/bookkeeper.conf` so you would need to check which settings it takes and which config file it populates, but I suspect you might be right about this one as well. @sijie can you confirm?

Reply:
You simply would be taking out -XX:+UseG1GC -XX:MaxGCPauseMillis=10 of the BOOKIE_MEM and putting it into BOOKIE_GC like so:

`BOOKIE_MEM: "\"-Xms15g -Xmx15g -XX:MaxDirectMemorySize=15g -Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.linkCapacity=1024 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB -XX:+ExitOnOutOfMemoryError -XX:+PerfDisableSharedMem -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintHeapAtGC -verbosegc -XX:G1LogLevel=finest\""`
`BOOKIE_GC: "\"-XX:+UseG1GC -XX:MaxGCPauseMillis=10\""`

`bookie-autorecovery` does run `bin/apply-config-from-env.py conf/bookkeeper.conf` so you would need to check which settings it takes and which config file it populates, but I suspect you might be right about this one as well. @sijie can you confirm?

Reply:
I will update soon after I test it.
In any case, "bin/apply-config-from-env.py conf/bkenv.sh" is definitely not needed (more so even bin/apply-config-from-env.py conf/pulsar_env.sh is not needed) as apply-config-from-env.py always updates the two sh files.

# Always apply env config to env scripts as well
conf_files = ['conf/pulsar_env.sh', 'conf/bkenv.sh'] + sys.argv[1:]

PF_ENV_PREFIX = 'PULSAR_PREFIX_'

for conf_filename in conf_files:

Reply:
@roman-popenov updated the file to set BOOKIE_MEM and BOOKIE_GC correctly both for bookie and for the autorecovery (validated that they are applied with jinfo)